### PR TITLE
add missing CloudProvider entry

### PR DESCRIPTION
**Pull request by fgimenez**
Related to https://github.com/giantswarm/giantswarm/issues/2281

### CODE REVIEW

**Comment by fgimenez** [comment]
Done PTAL

File: service/framework.go [hunk 1]
```diff
@@ -842,6 +845,23 @@ func migrateTPRsToCRDs(logger micrologger.Logger, clientSet *versioned.Clientset
 			} else if err != nil {
 				logger.Log("error", fmt.Sprintf("%#v", err))
 				return
+			} else if res.Spec.Cluster.Kubernetes.CloudProvider == "" {
+				// CRO existed with empty CloudProvider, refresh
+				patch := []v1alpha1.AWSConfig{}
+				patch[0].Spec.Cluster.Kubernetes.CloudProvider = awsCloudProvider
```
Comment (xh3b4sd) [inline service/framework.go]: I don't follow why we work with a list here when we actually only have one custom object per iteration. Is this some specialty of the PATCH magic?

Comment (fgimenez) [reply service/framework.go]: With `type.JSONPatchType` we need an array, `type.MergePatchType` accepts the JSON format of the required change directly (https://tools.ietf.org/html/rfc7386), which feels more natural, PTAL

**Comment by calvix** [comment]
What is the status on this?

**Comment by fgimenez** [comment]
Still needs more testing, the first approach didn't work, it was updating the already migrated TPOs, and these have the right cloud provider set after the migration fix. We need to query all the CROs and update those with empty cloud provider field. Also, the approach of `MergePatch` + specifying only the field to change didn't work, it erases the non-specified fields. So we need to give the complete CRO or use another patch type, which involves more testing.

The new clusters have the right cloud provider set, if there are existing clusters without it that we need to update quickly we could do that manually, pls let me know.

**Comment by xh3b4sd** [comment]
For my interest, when do we want to finish this? This was part of an incident situation and I thought we merge it the same day. Is there some problem which prevents us from making progress on this? It would be cool when we could get this sorted soon.

**Comment by fgimenez** [comment]
Sorry, I wasn't able to test on gauss yesterday, hopefully will be merged today, will keep you posted.

**Comment by fgimenez** [comment]
Successfully tested on gauss

**Comment by xh3b4sd** [comment]
🎉

**Pull request merged**